plugin/documents.py

```diff
@@ -528,7 +529,9 @@ def on_text_command(self, command_name: str, args: Optional[dict]) -> Optional[T
         return None
 
     def on_post_text_command(self, command_name: str, args: Optional[Dict[str, Any]]) -> None:
-        if command_name in ("next_field", "prev_field") and args is None:
+        if command_name == 'paste' and userprefs().format_on_paste:
+            self._did_paste = True
```

Comment:
It would kinda make sense to me that this functionality can be triggered on `paste_and_indent` (only or also) since it should work like an enhenced version of ST's native functionality.

Though not sure how this could be done.

I was thinking that there could be an LSP command that the user could assign custom keybinding to but that would have to work rather differently than what you did.

Alternatively the `format_on_save` could maybe be set to names of the commands that it should override so that one can set `paste_and_indent` only...

Reply:
https://github.com/sublimelsp/LSP/pull/2311#discussion_r1298568537

@jwortmann has the same comment.
 > > There exists also a command paste_and_indent (Ctrl+Shift+V).
Should it run the formatting too, when that command was invoked?
> 
>I would not add it, but don't have any strong opinions about it.
If you think that formatting should happen, I will add it.


Well you are the second person who asked the same question.
So I think I will add it.

> It would kinda make sense to me that this functionality can be triggered on paste_and_indent (only or also) since it should work like an enhenced version of ST's native functionality.

Are you suggesting the following:
If a user has `lsp_format_on_paste: false`,
and a user runs  `paste_and_indent`,
LSP should format the pasted range?

```diff
        if command_name == 'paste' and lsp_format_on_paste_enabled:
            self._did_paste = True
+        elif command_name == 'paste_and_indent':
+            self._did_paste = True
```
-----

Here was my proposal.
If a user has `lsp_format_on_paste: true`,
then regardless if the user run `paste` or `paste_and_indent`,
LSP will format the pasted range.

WDYT?


```diff
- if command_name == 'paste' and lsp_format_on_paste_enabled:
+ if command_name in ['paste', 'paste_and_indent'] and lsp_format_on_paste_enabled:
```

Reply:
The latter is what I meant but note that once `paste_and_ident` reformats things on its own, your code that tries to find pasted contents will likely fail...

Reply:
[c9d1b85](https://github.com/sublimelsp/LSP/pull/2397/commits/c9d1b852b55e41a166c73251da2e9a8be09842f2)

in this commit, I add code to intercept `paste_and_indent` and instead run the `paste` command.
> ... `paste_and_indent` reformats things on its own

as you noted,
I didn't want to add code to improve finding the reformatted text.
This approach didn't require a lot of code 
and while the intention is not that obvious obvious, I added a comment to explain why it was added.